lib/resolve-idx-refs.js

```diff
@@ -0,0 +1,26 @@
+'use strict'
+
+const scanner = require('object-scan')
+const toPath = require('lodash/toPath')
+const get = require('lodash/get')
+
+const findIdxRefs = (tree, selector, onRef) => {
```

Comment:
This is pretty hard to understand (same for the method below), can you add a comment that explains roughly what the different parameters mean?